File: DressupAndroid/app/src/main/java/cn/dreamfruits/yaoguo/module/main/diy/zrbannerpage/CardAdapter.java
//package cn.dreamfruits.yaoguo.module.main.diy.zrbannerpage;
//
//import android.view.LayoutInflater;
//import android.view.View;
//import android.view.ViewGroup;
//import android.widget.ImageView;
//
//import androidx.recyclerview.widget.RecyclerView;
//
//
//import java.util.ArrayList;
//import java.util.List;
//
//import cn.dreamfruits.yaoguo.R;
//import cn.dreamfruits.yaoguo.module.main.diy.zrbannerpage.library.CardAdapterHelper;
//
//
///**
// * Created by jameson on 8/30/16.
// */
//class CardAdapter extends RecyclerView.Adapter<CardAdapter.ViewHolder> {
//    private List<Integer> mList = new ArrayList<>();
//    private CardAdapterHelper mCardAdapterHelper = new CardAdapterHelper();
//
//    public CardAdapter(List<Integer> mList) {
//        this.mList = mList;
//    }
//
//    @Override
//    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
//        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_card_item, parent, false);
//        mCardAdapterHelper.onCreateViewHolder(parent, itemView);
//        return new ViewHolder(itemView);
//    }
//
//    @Override
//    public void onBindViewHolder(final ViewHolder holder, final int position) {
//        mCardAdapterHelper.onBindViewHolder(holder.itemView, position, getItemCount());
//        holder.mImageView.setImageResource(mList.get(position));
//        holder.mImageView.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View view) {
////                ToastUtils.show(holder.mImageView.getContext(), "" + position);
//            }
//        });
//    }
//
//    @Override
//    public int getItemCount() {
//        return mList.size();
//    }
//
//    public class ViewHolder extends RecyclerView.ViewHolder {
//        public final ImageView mImageView;
//
//        public ViewHolder(final View itemView) {
//            super(itemView);
//            mImageView = (ImageView) itemView.findViewById(R.id.imageView);
//        }
//
//    }
//
//}

File: DressupAndroid/app/src/main/java/cn/dreamfruits/yaoguo/module/main/diy/zrbannerpage/CardAdapter.kt
package cn.dreamfruits.yaoguo.module.main.diy.zrbannerpage

import android.annotation.SuppressLint
import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import cn.dreamfruits.yaoguo.R
import cn.dreamfruits.yaoguo.module.main.diy.zrbannerpage.library.CardAdapterHelper
import cn.dreamfruits.yaoguo.repository.bean.diy.GetStyleVersionListByTypeBean
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy

/**
 * @Author qiwangi
 * @Date 2023/7/11
 * @TIME 16:59
 */
class CardAdapter(
    private var mContent: Context,
    private var dataList: MutableList<GetStyleVersionListByTypeBean.Item>
) : RecyclerView.Adapter<CardAdapter.ViewHolder>() {
    private val mCardAdapterHelper = CardAdapterHelper()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView =
            LayoutInflater.from(parent.context).inflate(R.layout.view_card_item, parent, false)
        mCardAdapterHelper.onCreateViewHolder(parent, itemView)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        mCardAdapterHelper.onBindViewHolder(holder.itemView, position, itemCount)


        /**预加载回调*/
        if (dataList.size > 3 && position == dataList.size - 3) {
            mInterface!!.onPreload()
        }

        /**图片*/
        Glide.with(mContent)
            .asBitmap()
            .dontAnimate()
            .skipMemoryCache(false)
            .diskCacheStrategy(DiskCacheStrategy.ALL)
            .circleCrop()
            .error(R.drawable.temp_icon)
            .load(dataList[position].coverUrl)
            .into(holder.mImageView)

        /**标题*/
        holder.mTv.text = dataList[position].name


        holder.mImageView.setOnClickListener { view ->
            mInterface!!.onclick(dataList[position].id, position)
            Log.e(
                "zqr176543",
                "适配器中 当前点击了banner的   ======" + position + "" + dataList[position].name
            )
        }
    }

    override fun getItemCount(): Int {
        return dataList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val mImageView: ImageView = itemView.findViewById(R.id.imageView)
        val mTv: TextView = itemView.findViewById(R.id.zr_title)
    }


    /**
     * 设置数据
     */
    @SuppressLint("NotifyDataSetChanged")
    fun setData(tempList: MutableList<GetStyleVersionListByTypeBean.Item>, isClear: Boolean) {
        if (isClear) {
            this.dataList.clear()
            this.dataList = tempList
            notifyDataSetChanged()
        } else {
            this.dataList += tempList
            notifyItemRangeChanged(dataList.size, dataList.size + tempList.size)
        }
    }


    //回调
    interface InnerInterface {
        fun onclick(id: Long, position: Int)

        fun onPreload()
    }

    private var mInterface: InnerInterface? = null

    fun setCardAdapterCallBack(mInterface: InnerInterface) {
        this.mInterface = mInterface
    }

}